proto/rill/runtime/v1/api.proto

```diff
@@ -215,6 +219,24 @@ message PingResponse {
   google.protobuf.Timestamp time = 2;
 }
 
+message HealthzRequest {}
+
+// Response message for RuntimeService.Healthz
+// The fields will be set to corresponding errors if any.
+message HealthzResponse {
+  string limiter = 1;
+  string conn_cache = 2;
+  string metastore = 3;
+  string network = 4;
+  map<string, InstanceHealth> instances = 5;
+}
+
+message InstanceHealth {
+  string controller = 1;
+  string repo_sync = 2;
+  string admin_connect = 3;
+}
```

Comment:
Subjective, but I would consider putting an `_error` suffix on the string types here since the message names do not themselves imply that these are errors.